### PR TITLE
22440 medical copay initial backend integration

### PR DESCRIPTION
**Pull request by dillo**
## Description of change

22440 Medical Copays Integration
    - Setup route/endpoint for the UI
    - Setup medical_copays controller
    - Setup medical_copays service layer
    - Setup medical_copays service objects
    - Setup basic VBS integration
    - Setup basic configuration data in settings
    - Setup adequate test coverage
    - Feature is behind Flipper: `show_medical_copays`

The `settings.yml` and `test.yml` files have very basic settings for the Medical Copays VBS service. They will be populated once the downstream Liberty team provisions those details. The appropriate files in the DevOps repo will also be updated accordingly.
## Original issue(s)
department-of-veterans-affairs/va.gov-team#22440

## Things to know about this PR

- [x] Rubocop passing

### CODE REVIEW

File: app/services/medical_copays/request.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  #   @return (see Config::Options#host)
+  # @!attribute service_name
+  #   @return (see Config::Options#service_name)
+  # @!attribute url
+  #   @return (see Config::Options#url)
+  class Request
+    extend Forwardable
+    include Common::Client::Concerns::Monitoring
+
+    STATSD_KEY_PREFIX = 'api.medical_copays.request'
```
Comment (LindseySaari) [inline app/services/medical_copays/request.rb:19]: It looks like you'll need to init this statsd key in the statsd initializer

File: app/services/medical_copays/vbs/request_data.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        new(opts)
+      end
+
+      ##
+      # The schema for validating attribute data
+      #
+      # @return [Hash]
+      #
+      def self.statements_schema
+        {
```
Comment (LindseySaari) [inline app/services/medical_copays/vbs/request_data.rb:36]: Can the vets-json-schema repo be utilized for the schema validation?

File: config/settings.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 hostname: 127.0.0.1:3000 # possible fallback for unsafe request.host
 
-vsp_environment:
```
Comment (LindseySaari) [inline config/settings.yml:3]: This looks like some kind of typo/editor change?

File: config/settings.yml [hunk 4]
```diff
@@ -74,6 +74,13 @@ dmc:
   url: https://internal-dsva-vagov-dev-fwdproxy-1893365470.us-gov-west-1.elb.amazonaws.com:4465/api/v1/digital-services/
   debts_endpoint: debt-letter/get
 
+mcp:
```
Comment (LindseySaari) [inline config/settings.yml:77]: Will there be entries for these keys in devops repo, if so, could you provide the link in the PR desc?

File: app/services/medical_copays/vbs/service.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      # @return [Service] an instance of this class
+      #
+      def self.build(opts = {})
+        new(opts)
+      end
+
+      def initialize(opts)
+        @request = MedicalCopays::Request.build
+        @request_data = RequestData.build(user: opts[:user])
+        @response_data = ResponseData
```
Comment (LindseySaari) [inline app/services/medical_copays/vbs/service.rb:30]: Do you need to call `ResponseData.build `here

File: config/settings.yml (hunk 4 above)
Comment (dillo) [reply config/settings.yml:77]: Yes, the entries will be made in the DevOps repo once the Liberty team provides us the configuration data for the VBS service. For now, everything will be hidden behind a feature flip.

File: config/settings.yml (hunk 3 above)
Comment (dillo) [reply config/settings.yml:3]: I'm not entirely sure. I will check up on it!

Comment (dillo) [reply config/settings.yml:3]: Looks like my editor just stripped that additional trailing white space.

File: config/settings.yml (hunk 4 above)
Comment (dillo) [reply config/settings.yml:77]: I've updated the PR description to reflect my above comment.

File: app/services/medical_copays/request.rb (hunk 1 above)
Comment (dillo) [reply app/services/medical_copays/request.rb:19]: I think if I'm calling `with_monitoring` in the `post` method below I should be able to use what's provided in `lib/common/client/concerns/monitoring.rb` to log the stats? The `total` and `failure` stats is all I'm looking to capture for the MVP. For all custom monitoring I will be setting up the config in the stated initializer 👍

File: app/services/medical_copays/vbs/request_data.rb (hunk 2 above)
Comment (dillo) [reply app/services/medical_copays/vbs/request_data.rb:36]: I don't know what that repo is 😳  I'll have to look into it, thanks! However, I think what I have going on right now is a bit heavy-handed for validating strings and could probably be simplified.

File: app/services/medical_copays/vbs/service.rb (hunk 5 above)
Comment (dillo) [reply app/services/medical_copays/vbs/service.rb]: Moving the ResponseData and calling build on it directly inside the `get_copays` method. 👍

File: app/controllers/v0/medical_copays_controller.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module V0
+  class MedicalCopaysController < ApplicationController
+    before_action { authorize :medical_copays, :access? }
+
+    def index
```
Comment (LindseySaari) [inline app/controllers/v0/medical_copays_controller.rb:7]: Do you have swagger updates for this new endpoint?

Comment (dillo) [reply app/controllers/v0/medical_copays_controller.rb:7]: Those are coming up next 👍  (Separate PR)

**Review by LindseySaari — DISMISSED** [review]

**Comment by dillo** [comment]
@LindseySaari branch always seems to be out of date with master and I don't know if I can find a window of opportunity to merge.

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**